add search by ORCIDID method

peetucket opened the pull request:
## Why was this change made? 🤔

Fixes sul-dlss/happy-heron#3166

1. Add a (temporary implementation) of the search by ORCIDID method that just iterates over all results until it finds what we need and then return it.
2. Update README to indicate how to create new cassettes

## How was this change tested? 🤨

New tests

peetucket commented on lib/mais_orcid_client.rb:
```diff
@@ -57,19 +57,51 @@ def fetch_orcid_users(limit: nil, page_size: nil)
+
+    sunetid ? fetch_by_sunetid(sunetid) : fetch_by_orcidid(orcidid)
+  end
+
```
this existing method is extended to support fetching by sunetid or orcidid

the method signature remains identical for all consumers who search by sunetid, so changes needed for them

peetucket commented on lib/mais_orcid_client.rb:
```diff
@@ -57,19 +57,51 @@ def fetch_orcid_users(limit: nil, page_size: nil)
 
-  def first_page(page_size)
+  def first_page(page_size = nil)
     path = "/users?scope=ANY"
```
make this param optional, and just default to nil, which is supported by the method anyway (it just uses the API default)

justinlittman commented on README.md:
```diff
@@ -58,3 +58,20 @@ end
+
+## VCR Cassettes
+
+VCR gem is used to record the results of the API calls for the tests.  If you need to
```
VCR allows you to obscure automatically through configuration. (See sul_pub for example.)

peetucket replied:
Yeah, I know - some of the auto obfuscation requires you to specify the string to be obscured, which in sul-pub comes from Settings (which we don't have in the gem).  But I'll check if there is a more clever way to do this, like regex.

peetucket commented:
Added sanitization where I could, and updated README.  Couldn't figure out how to sanitize multiple mathces of the same regex in a single cassette though (which is needed for user access tokens)
